packages/cli/src/lib/setups/i18n/replacers.ts

```diff
@@ -342,3 +342,192 @@ async function findEntryFile({
 
   return {filepath, astType};
 }
+
+type FilePath = {
+  rootDirectory: string;
+  path: string;
+  filename: string;
+};
+
+async function findFile({rootDirectory, path, filename}: FilePath) {
+  if (!fileExists(joinPath(rootDirectory, path, filename))) return {};
+
+  const {filepath, astType} = await findFileWithExtension(
+    joinPath(rootDirectory, path),
+    filename,
+  );
+
+  return {filepath, astType};
+}
+
+export async function replaceI18nCartPath(
+  {rootDirectory}: I18nSetupConfig,
+  formatConfig: FormatOptions,
+) {
+  await Promise.all([
+    replaceI18nRootFile({rootDirectory}, formatConfig),
+    replaceI18nCartPathForFile(
+      {
+        rootDirectory,
+        path: 'app/components',
+        filename: 'Cart',
+      },
+      formatConfig,
+    ),
+    replaceI18nCartPathForFile(
+      {
+        rootDirectory,
+        path: 'app/components',
+        filename: 'AddToCartButton',
+      },
+      formatConfig,
+    ),
+  ]);
+}
+
+async function replaceI18nRootFile(
+  {rootDirectory}: I18nSetupConfig,
+  formatConfig: FormatOptions,
+) {
+  // Add selectedLocale to root loader
+  const {filepath, astType} = await findFile({
+    rootDirectory,
+    path: 'app',
+    filename: 'root',
+  });
+
+  if (!filepath || !astType) {
+    throw new AbortError('Could not find a root file');
+  }
+
+  await replaceFileContent(filepath, formatConfig, async (content) => {
+    const astGrep = await importLangAstGrep(astType);
+    var root = astGrep.parse(content).root();
+    const loaderReturnNode = root.find({
+      rule: {
+        kind: 'shorthand_property_identifier',
+        regex: 'publicStoreDomain',
```

Comment:
Again, since this code could run for existing user projects, we could make this more robust. There's an example that finds the return statement from the `links` export here: https://github.com/Shopify/hydrogen/blob/24819065b67e268be55efe0fb02d0bd3ed17996e/packages/cli/src/lib/setups/css/replacers.ts#L144-L174

 You can use the same thing to find the return statement of the loader export instead, and then add the new property.